CONTRIBUTING.md

```diff
@@ -56,19 +56,49 @@ disclosure of security bugs. In those cases, please go through the process
 outlined on that page and do not file a public issue.
 
 
+## Set up your development environment
+
+We use [uv](https://github.com/astral-sh/uv) to manage python dependencies and virtual environments.
+You can install `uv` by following this [guide](https://docs.astral.sh/uv/getting-started/installation/).
+You can install the dependencies by running:
+
+```bash
+$ cd llama-stack
+$ uv sync --extra dev
+$ uv pip install -e .
+$ source .venv/bin/activate
+```
+
 ## Pre-commit Hooks
 
 We use [pre-commit](https://pre-commit.com/) to run linting and formatting checks on your code. You can install the pre-commit hooks by running:
 
 ```bash
-$ cd llama-stack
-$ conda activate <your-environment>
-$ pip install pre-commit
-$ pre-commit install
+$ uv run pre-commit install
 ```
 
 After that, pre-commit hooks will run automatically before each commit.
 
+Alternatively, if you don't want to install the pre-commit hooks, you can run the checks manually by running:
+
+```bash
+$ uv run pre-commit run --all-files
+```
+
+## Adding a new dependency to the project
+
+To add a new dependency to the project, you can use the `uv` command. For example, to add `foo` to the project, you can run:
+
+```bash
+$ uv add foo
+$ uv sync
+```
+
+Finally sync the `requirements.txt` file by running:
```

Comment:
This should not be necessary to do manually because it is handled automatically by the pre-commit hook

Reply:
Ok I'm removing this but I'll leave an indication that `pre-commit` must run before committing.